src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -247,8 +250,13 @@ def get_path_to_port_config_file(hwsku=None, asic=None):
 
     port_config_candidates = []
 
-    # Check for 'platform.json' file presence first
-    port_config_candidates.append(os.path.join(platform_path, PLATFORM_JSON_FILE))
+    # Check for 'hwsku.json' file presence first
+    hwsku_json_file = os.path.join(hwsku_path, HWSKU_JSON_FILE)
+
+    # if 'hwsku.json' file is available, Check for 'platform.json' file presence,
+    # if 'platform.json' is available, APPEND it. Otherwise, SKIP it.
+    if os.path.isfile(hwsku_json_file):
```

Comment:
Do you need to check both platform file and hwsku file? Is platform file checked earlier or later? What if the platform has hwsku file but not platform file?

Reply:
it was checked below: 
`if os.path.isfile(candidate):`

We could completely reconstruct the code and return files directly, but for now, following the old structure should work too.